distiller.js

```diff
@@ -383,6 +384,61 @@ export class DataChunks {
     });
   }
 
+  /**
+   * Adds a cluster facet, derived from an existing facet. This facet
+   * will group the data into clusters based on the URL paths.
+   * You can specify the number of clusters and a producer function to
+   * generate the clusters.
+   * @param {string} facetName name of your new facet
+   * @param {string} baseFacet name of the base facet, from which to derive the clusters
+   * @param {object} clusterOptions options
+   * @param {number} clusterOptions.count number of clusters, The default value is log10(nValues)
+   * @param {function} clusterOptions.producer function that takes the cluster value and returns all possible cluster values
+   */
+  addClusterFacet(facetName, baseFacet, { count: clustercount = Math.floor(Math.log10(this.facets[baseFacet].length)) }) {
+    const facetValues = this.facets[baseFacet];
+
+    const createClusterMap = () => {
+      const clusterMap = facetValues.reduce((map, facet) => {
+        const clusters = urlProducer(facet.value);
+        clusters.forEach(cluster => {
+          if (!map.has(cluster)) {
+            map.set(cluster, 0);
+          }
+          map.set(cluster, map.get(cluster) + 1);
+        });
+        return map;
+      }, new Map());
+
+      // Find the most occurring cluster
+      let mostOccurringCluster = null;
+      let maxCount = 0;
+
+      for (const [cluster, count] of clusterMap.entries()) {
+        if (count > maxCount) {
+            maxCount = count;
+            mostOccurringCluster = cluster;
+        }
+      }
+
+      // Calculate the total number of URLs in the superset cluster
+      const totalUrlsInSupersetCluster = Math.floor(facetValues.length + clustercount);
+      return { clusterMap, mostOccurringCluster, totalUrlsInSupersetCluster };
+    };
+
+    const { clusterMap, mostOccurringCluster, totalUrlsInSupersetCluster } = createClusterMap();
+    const sortedClusters = [...clusterMap.entries()]
+        .sort((a, b) => b[1] - a[1])
+        .slice(0, clustercount)
+        .map(([cluster]) => cluster);
+
+    this.addFacet(facetName, (bundle) => {
+        const facetMatch = facetValues.find(f => f.entries.some(e => e.id === bundle.id));
+        const clusters = urlProducer(facetMatch.value);
+        return { facetMatch, ...clusters.filter(cluster => sortedClusters.includes(cluster)) };
```

Comment:
```suggestion
        return [ facetMatch, ...clusters.filter(cluster => sortedClusters.includes(cluster)) ];
```

facet functions return arrays, not objects.